shanoir-uploader/src/main/java/org/shanoir/uploader/action/SelectionActionListener.java

```diff
@@ -56,72 +56,68 @@ public void valueChanged(TreeSelectionEvent event) {
 		dicomData = null;
 
 		// clean up editPanel when value changed
-		mainWindow.noAnonR.setEnabled(false);
-		mainWindow.yesAnonR.setEnabled(false);
 		mainWindow.lastNameTF.setEnabled(false);
 		mainWindow.birthNameCopyButton.setEnabled(false);
 		mainWindow.firstNameTF.setEnabled(false);
 		mainWindow.birthNameTF.setEnabled(false);
 		mainWindow.birthDateTF.setEnabled(false);
-		mainWindow.newPatientIDTF.setEnabled(false);
 		mainWindow.mSexR.setEnabled(false);
 		mainWindow.fSexR.setEnabled(false);
 		mainWindow.lastNameTF.setText("");
 		mainWindow.firstNameTF.setText("");
 		mainWindow.birthNameTF.setText("");
 		mainWindow.birthDateTF.setText("");
-		mainWindow.newPatientIDTF.setText("");
 
 		mainWindow.isDicomObjectSelected = true;
-		selectedSeries = new LinkedHashSet<Serie>();
+		selectedSeries = new LinkedHashSet<SerieTreeNode>();
 
 		// check if multiple subject have been selected and show error
-		Patient patientAlreadySelected = null;
+		PatientTreeNode patientAlreadySelected = null;
 		// check if multiple studies have been selected and show error
 		Boolean oneStudyIsSelected = true;
 		// save the correct study selected
-		Study selectedStudy = null;
+		StudyTreeNode selectedStudy = null;
 		// returns all selected paths, which can be patients, studies and/or series
 		TreePath[] paths = mainWindow.dicomTree.getSelectionModel().getSelectionPaths();
 		if (paths != null && paths.length > 0) {
 			for (int i = 0; i < paths.length; i++) {
 				TreePath tp = paths[i];
 				Object o = tp.getLastPathComponent();
-				Patient patient = null;
+				PatientTreeNode patient = null;
 				// handle if patient in paths has been found
-				if (o instanceof Patient) {

```

Comment:
It seems in Java 17 you can now switch case with instanceof (and avoid explicit casting) : https://www.baeldung.com/java-switch-pattern-matching#1-type-pattern